src/jobs/jobs.controller.ts

```diff
@@ -475,25 +554,26 @@ export class JobsController {
   async create(
     @Req() request: Request,
     @Body() createJobDto: CreateJobDto,
-  ): Promise<JobClass> {
+  ): Promise<JobClass | null> {
     Logger.log("Creating job!");
     // Validate that request matches the current configuration
-    await this.validateJob(createJobDto);
+    const jobConfigVersion = await this.validateJob(createJobDto);
     // Check job authorization
-    await this.instanceAuthorization(createJobDto.jobParams, request.user as JWTUser);
+    await this.instanceAuthorization(createJobDto, request.user as JWTUser, "create");
     // Create actual job in database
-    const createdJobInstance = await this.jobsService.create(createJobDto);
+    const createdJobInstance = await this.jobsService.create(createJobDto, jobConfigVersion);
     // Perform the action that is specified in the create portion of the job configuration
     await this.performJobCreateAction(createdJobInstance);
-    // TBD
-    // Update job instance with results of job create action
-    // return await this.jobsService.statusUpdate(createdJobInstance);
-    return createdJobInstance;
+    // Update job instance after the success of job create action
+    return await this.jobsService.statusUpdate(
+      createdJobInstance.id,
+      { "jobStatusMessage": "Job has been created.", "jobStatusCode": "jobCreated" } as UpdateJobStatusDto
```

Comment:
Why do this as a statusUpdate rather than part of the create call? I suspect that statusUpdate could have side effects like emitting events to any registered listeners. This could cause some actions to run twice.

Even without side effects, I think we want the job to have the correct statusCode before `performJobCreateAction`, so this should move up a line.

Reply:
I am removing this part. Now message gets set to `jobCreated` before `performJobCreateAction`